### ISSUE TITLE
Impossible to export in CSV when HTML in data

### ISSUE THREAD

**Issue by ghost** [sequelpro/sequelpro#979]
Issue imported from Google Code: https://code.google.com/p/sequel-pro/issues/detail?id=979
Reporter:hcam...@gmail.com Date:2011-02-14 08:56:42 Status:New
## What steps will reproduce the problem?
1. Export one table in CSV in any format (/t , ;...) with no escape caracter, as will need not to escape special characters in HTML
## What is the expected output?

One record per line.
## What do you see instead?

One record on several lines. Impossible to use/import in any soft : excel, filemaker, bento....
Even tried /n /r... same problem.
## What version of Sequel Pro are you using?

0.9.8.1 build 2492
0.9.8.1 build 3193 (also will hang when choosing no escape character)
## What version of MySQL are you connecting to on the server?

5.5.9
## <b>Please provide any additional information below.</b>

Hope you can help on this. Thanks

Hubert CAMPAN

**Comment by bashu** [comment]
+1

**Comment by andrezimpel** [comment]
+1

**Comment by DfKimera** [comment]
This issue still persists, several years later :/

**Comment by abhibeckert** [comment]
@DfKimera thanks for the bump, I'll look into this soon.

**Comment by abhibeckert** [comment]
As far as I can tell, the problem is we use a backslash as the escape character.

By simply changing the escape character to a double quote, CSV export for HTML works perfectly:

![screen shot 2016-06-03 at 16 03 35](https://cloud.githubusercontent.com/assets/266088/15770205/0c4a322a-29a5-11e6-9ceb-bf0ce28e764b.png)

Does this work for you @DfKimera?

**Comment by DfKimera** [comment]
@abhibeckert Yep, it works! It seems slightly counter-intuitive, though. Shouldn't Sequel Pro detect the presence of the escape character in a cell's content and escape it before exporting?

**Comment by abhibeckert** [comment]
@DfKimera CSV doesn't have a proper standard, and in my experience it's not possible to create a CSV file that works everywhere.

Our default escape character (backslash) doesn't work properly with Microsoft Excel (or Apple Numbers), but it may be required elsewhere. I'll have to do a bit of research.

**Comment by dmoagx** [comment]
@abhibeckert There actually is a standard for CSV https://tools.ietf.org/html/rfc4180 :)

**Comment by abhibeckert** [comment]
@dmoagx "This memo provides information [...]  It does
   not specify an Internet standard of any kind."

If you follow that standard, you'll create files that cannot be opened in Excel for example. And a file compatible with Excel will not be compatible with any software that uses the RFC.

Also, Excel chokes on fields that with lengthy strings of digits (such as phone numbers), and the way it applies locale settings to decimal place is a disaster.

Instead of following the RFC, we should try to be compatible with MySQL's `load data infile` system - which is the fastest way to perform bulk-write operations on an InnoDB table.

**Comment by fulldecent** [comment]
See RFC 4180

https://tools.ietf.org/html/rfc4180

**Comment by abhibeckert** [comment]
@fulldecent see comments here from a few weeks ago.

That's only a "request for comment", not a standard, and it is not compatible with Excel - so unlikely to become a standard unless that changes.
